src/StreamController.h

```diff
@@ -61,6 +64,16 @@ class StreamController : public IStreamController {
   /// last message to be written in nanoseconds.
   void setStopTime(const time_point &StopTime) override;
 
+  /// \brief Pause consumers.
+  ///
+  /// Pauses consumer polling to throttle the ingestion of data.
+  void pauseStreamers() override final;
```

Comment:
I'd probably not bother marking these methods as final. Very few other methods seem to have been marked final, so marking these two as final could be read as meaning there is something special about these two methods which I don't think is the case.